agent/resource.go

```diff
@@ -87,6 +101,55 @@ func (a *Agent) processIncomingResourceRequest(ev *event.Event) error {
 	return nil
 }
 
+func (a *Agent) processIncomingPostResourceRequest(ctx context.Context, req *event.ResourceRequest, gvr schema.GroupVersionResource) (*unstructured.Unstructured, error) {
```

Comment:
Is Argo CD actually using POST to create new resources?

Reply:
Not directly. But Argo CD calls [Create()](https://github.com/argoproj/argo-cd/blob/2c3a452fd7514777deb98679f68586242b9feaa7/server/application/application.go#L2509) while executing the resource action that gets translated to a [POST request](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.33/#create-deployment-v1-apps).